jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/AbstractDeploymentService.java

```diff
@@ -176,19 +179,46 @@ public void commonDeploy(DeploymentUnit unit, DeployedUnitImpl deployedUnit, Run
     
     @Override
     public void undeploy(DeploymentUnit unit) {
+        undeploy(unit, false);
+    }
+
+    @Override
+    public void undeploy(DeploymentUnit unit, boolean abortInstances) {
+
         List<Integer> states = new ArrayList<Integer>();
         states.add(ProcessInstance.STATE_ACTIVE);
         states.add(ProcessInstance.STATE_PENDING);
         states.add(ProcessInstance.STATE_SUSPENDED);
         Collection<ProcessInstanceDesc> activeProcesses = runtimeDataService.getProcessInstancesByDeploymentId(unit.getIdentifier(), states, new QueryContext());
-        if (!activeProcesses.isEmpty()) {
-            throw new IllegalStateException("Undeploy forbidden - there are active processes instances for deployment " 
-                                            + unit.getIdentifier());
+        if (!abortInstances && !activeProcesses.isEmpty()) {
+            throw new IllegalStateException("Undeploy forbidden - there are active processes instances for deployment "
+                                                    + unit.getIdentifier());
         }
+
         synchronized (this) {
             DeployedUnit deployed = deploymentsMap.remove(unit.getIdentifier());
+
             if (deployed != null) {
+
                 RuntimeManager manager = deployed.getRuntimeManager();
+
+                if(abortInstances && !activeProcesses.isEmpty()) {
```

Comment:
Formatting..
```suggestion
                if (abortInstances && !activeProcesses.isEmpty()) {
```

Reply:
You don't have to reformat the whole class, but adding spaces after if / for etc shouldn't need reformatting the whole class. Up to you..